### PR TITLE
Remove pyopenssl cruft

### PR DESCRIPTION
**Pull request by KarlParkinson**
#### Description

Addresses #4831.

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: mitmproxy/net/tls.py [hunk 1]
```diff
@@ -21,29 +21,21 @@
 
 # redeclared here for strict type checking
 class Method(Enum):
-    # TODO: just SSL attributes once https://github.com/pyca/pyopenssl/pull/985 has landed.
-    TLS_SERVER_METHOD = getattr(SSL, "TLS_SERVER_METHOD", 8)
-    TLS_CLIENT_METHOD = getattr(SSL, "TLS_CLIENT_METHOD", 9)
+    TLS_SERVER_METHOD = getattr(SSL, "TLS_SERVER_METHOD")
+    TLS_CLIENT_METHOD = getattr(SSL, "TLS_CLIENT_METHOD")
```
Comment (mhils) [inline mitmproxy/net/tls.py:25]: We don't need to use `getattr` here or below, we can just use `SSL.TLS_SERVER_METHOD` directly. :)

File: mitmproxy/net/tls.py [hunk 2]
```diff
@@ -21,29 +21,21 @@
...
 
 
-# TODO: remove once https://github.com/pyca/pyopenssl/pull/985 has landed.
```
Comment (mhils) [inline mitmproxy/net/tls.py:29]: I think the comment here got moved somehow. Let's keep the version check, but remove the `setdefault` lines below. They shouldn't be necessary.

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@
 * Fix a crash caused when editing string option (#4852, @rbdixon)
 * Base container image bumped to Debian 11 Bullseye (@Kriechi)
 * Upstream replays don't do CONNECT on plaintext HTTP requests (#4876, @HoffmannP)
+* Remove old pyOpenSSL cruft (#4831, @KarlParkinson)
```
Comment (mhils) [inline CHANGELOG.md:17]: Let's call this "remove workarounds for old pyOpenSSL versions", in retrospect I realize that my wording from the issue could be misunderstood. pyOpenSSL is not crufty. :)

**Review by mhils — COMMENTED** [review]
Thanks for taking care of this! 🍰

**Review by mhils — APPROVED** [review]
Fantastic, thanks! 🍰

**Comment by KarlParkinson** [comment]
My pleasure @mhils.

**Pull request merged**